### DISCUSSION TITLE
Repeated dynamic blob-generation and automatic USDZ conversion.

### DISCUSSION THREAD

**Discussion by Skyswimsky** [google/model-viewer#2975] [Q&A]
First off, English isn't my native language and it's kind of the first time I am posting something like this. So apologies if I screw something up.

I'm working on a project where the user creates dynamic models in three.js and can then look at them via AR, both export into the proper file format(gltf) and launch of AR is done with the click of a single button. I let model-viewers automatic USDZ conversion take care of iOS as the blob doesn't seem to work with the ios-src and quick-look.

The problem occurs when the user returns from quick-look, makes some changes to the model and then wants start again. While the proper android src is set, it still shows the old model for the iOS device. I have super limited debug capabilities for iOS myself.

I tried removing the ios-src as I thought maybe it's automatically set on the first time and thus doesn't update to the 'new model', but that isn't working.

Does anyone have any suggestions, please?

**Comment by uprealdigital** [comment]
That's a pretty interesting problem since I'm working on something similar where 3d models are dynamically created. Curious about what people have to say even though I don't have an answer to that myself.

EDIT: I'm actually seeing the issue right now just changing the glb file manually, the auto-generated usdz file doesn't get updated. This might be a caching issue... I'm looking into it.

EDIT 2: I tried deleting the website caches and deleting the history but I'm still getting the same old geometry. Not sure how to force Safari to update the USDZ model.

**Reply by samaneh-kazemi** [reply]
@Skyswimsky Hi, thanks for reaching out. Can you please share a repro link so we can look into it?

**Reply by Skyswimsky** [reply]
@samaneh-kazemi Hi, sorry for the late reply. I've created a demo project with the necessary elements at https://github.com/Skyswimsky/threejs_modular_blob_export_modelviewer_demo I can not test if it works on mobile until next week, however. I hope I did everything properly, let me know if you need more!

**Reply by elalish** [reply]
@Skyswimsky Thanks, but could you  provide a link to the deployed URL of the site  that  demonstrates the problem? I don't want to build  your repo myself 😂

We do cache models, and we do it by their  URL. If you're making changes to the GLB outside of our scene graph API, I think you'll need to update the URL (maybe append a version  number) so that MV knows it's new.

**Reply by Skyswimsky** [reply]
@elalish Here is a link to a Codesandbox with the problem: https://codesandbox.io/s/charming-chatelet-5j1cg?file=/src/index.js

And the URL of the Object changes at runtime, as it'll always generate a new ObjectURL when you start.

**Reply by elalish** [reply]
@Skyswimsky Why are you using three.js directly in addition to model-viewer? model-viewer already includes all the functionality you're writing; the whole point of our library is to save you from having to use three.js directly by abstracting it at a higher level.

**Reply by Skyswimsky** [reply]
@elalish Because I need it to create modular objects at runtime that are then exported. Using the Model-Viewer functionality to display the object in AR for iOS and Android devices.

**Reply by elalish** [reply]
Your  code doesn't seem to interact with MV, so  I think you'd be better off asking on the three.js forums. If you want to change  an object's dimensions, export it and see it in AR, you  can do all of that  with MV directly, in which case we can  help. If you  have  a more complicated use case in  mind, please describe it, as this  is how we figure out  what to add to our API.
